### PR TITLE
Fix Gram matrix

### PR DESCRIPTION
**Pull request by Tom-Ren**
### Change the calculation of gram matrix

- Change `c, n = x.shape[1], x.size // x.shape[1]` into `c, n = x.shape[1], x.shape[2] * x.shape[3]` in `chapter_computer-vision/neural-style.md`,  Which is easy to understand .
- Add a link to explain gram matrix

### CODE REVIEW

**Comment by astonzhang** [comment]
Thanks. Can you first reverse your changes of chapter_computer-vision/image-augmentation.md and utils.py in this PR? The goal is, this PR only contains changes for neural-style

File: chapter_optimization/adagrad.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ $$\boldsymbol{x} \leftarrow \boldsymbol{x} - \boldsymbol{g}'.$$
 
 ## Adagrad的特点
 
-需要强调的是，小批量随机梯度按元素平方的累加变量$\boldsymbol{s}$出现在含调整后学习率的梯度$\boldsymbol{g}'$的分母项。因此，如果目标函数有关自变量中某个元素的偏导数一直都较大，那么就让该元素的学习率下降快一点；反之，如果目标函数有关自变量中某个元素的偏导数一直都较小，那么就让该元素的学习率下降慢一点。然而，由于$\boldsymbol{s}$一直在累加按元素平方的梯度，自变量中每个元素的学习率在迭代过程中一直在降低（或不变）。所以，当学习率在迭代早期降得较快且当前解依然不佳时，Adagrad在迭代后期由于学习率过小，可能较难找到一个有用的解。
+需要强调的是，小批量随机梯度按元素平方的累加变量$\boldsymbol{s}$出现在含调整后学习率的梯度$\boldsymbol{g}'$的分母项。因此，如果目标函数相对于自变量中某个元素的偏导数一直都较大，那么就让该元素的学习率下降快一点；反之，如果目标函数相对于自变量中某个元素的偏导数一直都较小，那么就让该元素的学习率下降慢一点。然而，由于$\boldsymbol{s}$一直在累加按元素平方的梯度，自变量中每个元素的学习率在迭代过程中一直在降低（或不变）。所以，当学习率在迭代早期降得较快且当前解依然不佳时，Adagrad在迭代后期由于学习率过小，可能较难找到一个有用的解。
```
Comment (astonzhang) [inline chapter_optimization/adagrad.md]: 这里我还是倾向于之前的写法，现在的改动意义变了。

Comment (Tom-Ren) [reply chapter_optimization/adagrad.md]: 好的，原本在读的时候觉得元素*elements*和变量*variables*有些没理解, 觉得偏导数是对于变量$X$中某个分量$x_i$来说的，所以权衡了一下改成这样了。

Comment (astonzhang) [reply chapter_optimization/adagrad.md]: 这里自变量中每个元素可以视为单独的变量，所以才表达为“目标函数有关自变量中某元素的偏导数”

File: chapter_computer-vision/neural-style.md [hunk 2]
```diff
@@ -118,11 +118,12 @@ def content_loss(y_hat, y):
...
+匹配统计分布常用的做法是冲量匹配，就是说使得他们有一样的均值，协方差，和其他高维的冲量。为了计算简单起见，我们只匹配二阶信息，即协方差。下面定义如何计算协方差矩阵(也称为[格拉姆矩阵](https://en.wikipedia.org/wiki/Gramian_matrix)),输出为 $c \times c$ 的矩阵：
 
 ```{.python .input  n=11}
 def gram(x):
-    c, n = x.shape[1], x.size // x.shape[1]
+    #c, n = x.shape[1], x.size // x.shape[1]
+    c, n = x.shape[1], x.shape[2] * x.shape[3]    #x.shape = (batch,channels,h,w)
     y = x.reshape((c, n))
     return nd.dot(y, y.T) / n
 ```
```
Comment (astonzhang) [inline chapter_computer-vision/neural-style.md]: @mli  Would you like to take a look at this?

**Comment by astonzhang** [comment]
I have merged the other pr.

**Pull request closed without merging**